File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int sum = 0;
        int i = 0;
        int j = 0;
        int h = mat.size()-1;
        while(i < mat.size())
        {
            sum += mat[i][j];
            if(i != h)
                sum += mat[h][j];
            i++;
            j++;
            h--;
        }
        return sum;
    }
};

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        sum = 0
        i = 0
        j = 0
        h = len(mat[0])-1
        # print(len(mat[0]))
        while(i < len(mat[0])):
            sum += mat[i][j]
            if(i != h):
                sum+=mat[h][j]
            i += 1
            j += 1
            h -= 1
        return sum